### ISSUE TITLE
AttributeError: Attempted to set type to proximal_regularizer, but CfgNode is immutable.Could please help me to overcome this problem,which happens when I call for the fedprox algorithm:python federatedscope/main.py --cfg federatedscope/example_configs/femnist.yaml fedprox.use True

### ISSUE THREAD

**Issue by Zst0514** [alibaba/FederatedScope#122]

**Comment by DavdGao** [comment]
Thanks for your trying. It seems like the `cfg` in unchangable, your can try the following code in `federatedscope/core/trainers/trainer_nbafl.py`
```
    cfg.regularizer.defrost()
    cfg.regularizer.type = 'proximal_regularizer'
    cfg.regularizer.mu = cfg.fedprox.mu
    cfg.regularizer.freeze(save=False)
```
, and we will fix this problem quickly.

**Comment by xieyxclack** [comment]
This thread is closed as we fix this problem in #124 
Thanks again for your comments!

**Comment by Zst0514** [comment]
I was a little busy the other day, sorry for replying to your email just now, thank you very much for your answer!I may ask you for more questions related in the future.













At 2022-06-02 11:45:32, "Yuexiang XIE" ***@***.***> wrote:

Closed #122 as completed.
